pycalphad/tests/test_database.py

```diff
@@ -64,9 +77,59 @@ def test_load_from_string():
 def test_export_import():
     "Equivalence of re-imported database to original."
     test_dbf = Database(ALNIPT_TDB)
-    assert Database.from_string(test_dbf.to_string(fmt='tdb'), fmt='tdb') == test_dbf
+    assert Database.from_string(test_dbf.to_string(fmt='tdb', if_incompatible='ignore'), fmt='tdb') == test_dbf
     test_dbf = Database(ALFE_TDB)
-    assert Database.from_string(test_dbf.to_string(fmt='tdb'), fmt='tdb') == test_dbf
+    assert Database.from_string(test_dbf.to_string(fmt='tdb', if_incompatible='ignore'), fmt='tdb') == test_dbf
+
+@nose.tools.raises(ValueError)
+def test_tc_incompatible_raises_error_by_default():
+    "Symbol names too long for Thermo-Calc raise error on write."
+    test_dbf = Database.from_string(INVALID_TDB_STR, fmt='tdb')
+    test_dbf.to_string(fmt='tdb')
+
+def test_tc_incompatible_warns_with_kwarg_warn():
+    "Symbol names too long for Thermo-Calc warn and write the database as given."
+    test_dbf = Database.from_string(INVALID_TDB_STR, fmt='tdb')
+    with warnings.catch_warnings(record=True) as w:
+        invalid_dbf = test_dbf.to_string(fmt='tdb', if_incompatible='warn')
+        assert len(w) > 0
+    assert test_dbf == Database.from_string(invalid_dbf, fmt='tdb')
+
+def test_tc_incompatible_ignores_with_kwarg_ignore():
+    "Symbol names too long for Thermo-Calc are ignored the database written as given."
+    test_dbf = Database.from_string(INVALID_TDB_STR, fmt='tdb')
+    with warnings.catch_warnings(record=True) as w:
+        invalid_dbf = test_dbf.to_string(fmt='tdb', if_incompatible='ignore')
+        assert len(w) == 0
+    assert test_dbf == Database.from_string(invalid_dbf, fmt='tdb')
+
+def test_tc_incompatible_mangles_names_with_kwarg_fix():
+    "Symbol names too long for Thermo-Calc are mangled and put in a JSON in the same dir as the TDB."
```

Comment:
The JSON is some cruft and should be removed. I was thinking that a possible solution would be to write a JSON file with the mangled names (e.g. symbol_name_ map to a JSON in the same directory as the TDB) then we could reconstitute with possibly a special keyword arg, but I didn't implement it. Another idea I had for reconstituting was to write somewhere constant relative to the package/os, but that wouldn't be portable. 

Ideally we could just use some comments, but comments are filtered in the read_tdb, so we can't automatically constitute.